### PR TITLE
feat: use credit class metadata in addition to sanity credit class data

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: regen-network/regen-registry#1428

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

C04 is not on Sanity staging but has some image stored in its metadata (as well as required name and description), see for instance:
https://deploy-preview-1935--regen-registry.netlify.app/credit-classes/C04
or https://deploy-preview-1935--regen-registry.netlify.app/project/C04-001
or any of the ecocredits / sell orders table.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/lib/ecocredit/api.ts [hunk 1]
```diff
@@ -290,22 +310,54 @@ const getClassProjectForBatch = async (
...
+  if (reactQueryClient) {
+    classData = await getFromCacheOrFetch({
+      query: getClassQuery({
+        request: { classId },
+        client: ecocreditClient,
+      }),
+      reactQueryClient,
+    });
+  } else {
+    // TODO Keeping the old path for retro-compatibility of the function.
```
Comment (flagrede) [inline web-registry/src/lib/ecocredit/api.ts:333]: Here the task I was referring to for that: https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/gh/regen-network/regen-registry/1539

**Review by flagrede — APPROVED** [review]
Looks good, nice job on the refactors!

**Comment by blushi** [comment]
@erikalogie see testing instructions above

**Comment by erikalogie** [comment]
@blushi This looks good but I'm not seeing the image on the credit class page: 
image

Otherwise it seems to be functioning as intended.

**Comment by blushi** [comment]
@erikalogie good catch, fixed now!

**Comment by erikalogie** [comment]
> @erikalogie good catch, fixed now!

great!

**Pull request merged**